src/main/java/ru/urururu/sanity/cpp/SourceRangeFactory.java

```diff
@@ -11,21 +12,28 @@
  * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
  */
 @Component
-public class SourceRangeFactory {
+public class SourceRangeFactory implements ParserListener {
     public static final int DW_TAG_file_type = 786473;
     public static final int DW_TAG_lexical_block = 786443;
     public static final int DW_TAG_subprogram = 786478;
 
     private static final int FILE_INDEX = 0;
     private static final int DIRECTORY_INDEX = 1;
 
+    private Long debugVersion;
+    private Byte versionByte;
+
     public SourceRange getSourceRange(SWIGTYPE_p_LLVMOpaqueValue instruction) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "getSourceRange" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)